### ISSUE TITLE
Extent wrong if layer contains point at 0,0

### ISSUE THREAD

**Issue by wonder-sk** [qgis/QGIS#33823]
I have a simple shapefile with three points, one of them has coordinates 0,0 - that one is completely ignored and extent is reported as QgsRectangle(0 1000, 1000 1000) instead of QgsRectangle(0 0, 1000 1000).

Testing data: [billboard-points.zip](https://github.com/qgis/QGIS/files/4065726/billboard-points.zip)

Somehow related, the Information tab in layer properties says that `Extent: Empty` which is not quite true (it may have zero height, but that does not mean it is completely empty).

**Comment by nyalldawson** [comment]
It's a long term silliness in QgsRectangle -- it considers a rectangle with x/y min max == 0/0 as a "null" rectangle and ignores it. This needs to be changed by adding a proper "invalid" QgsRectangle state (as was done recently for QgsPoint/QgsPointXY)

**Comment by elpaso** [comment]
Yes, it hit me too a few times. Can we do anything now or is it too risky?

**Comment by lbartoletti** [comment]
+1.

> Can we do anything now or is it too risky?

Maybe with QgsQuadrilateral class?

**Comment by rouault** [comment]
This was actually a OGR Shapefile driver bug that has been fixed in GDAL 3.1.0 : https://github.com/OSGeo/gdal/issues/2269

```
$ unzip  billboard-points.zip 

l$ ogrinfo billboard-points.shp -al
INFO: Open of `billboard-points.shp'
      using driver `ESRI Shapefile' successful.

Layer name: billboard-points
Metadata:
  DBF_DATE_LAST_UPDATE=2020-01-13
Geometry: 3D Point
Feature Count: 3
Extent: (0.000000, 1000.000000) - (1000.000000, 1000.000000)
Layer SRS WKT:
(unknown)
FID: Integer64 (11.0)
OGRFeature(billboard-points):0
  FID (Integer64) = 0
  POINT Z (0 0 50)

OGRFeature(billboard-points):1
  FID (Integer64) = 1
  POINT Z (0 1000 100)

OGRFeature(billboard-points):2
  FID (Integer64) = 2
  POINT Z (1000 1000 200)

$ ogrinfo billboard-points.shp -sql "recompute extent on billboard-points"
INFO: Open of `billboard-points.shp'
      using driver `ESRI Shapefile' successful.

$ ogrinfo billboard-points.shp -al
INFO: Open of `billboard-points.shp'
      using driver `ESRI Shapefile' successful.

Layer name: billboard-points
Metadata:
  DBF_DATE_LAST_UPDATE=2020-01-13
Geometry: 3D Point
Feature Count: 3
Extent: (0.000000, 0.000000) - (1000.000000, 1000.000000)
Layer SRS WKT:
(unknown)
FID: Integer64 (11.0)
OGRFeature(billboard-points):0
  FID (Integer64) = 0
  POINT Z (0 0 50)

OGRFeature(billboard-points):1
  FID (Integer64) = 1
  POINT Z (0 1000 100)

OGRFeature(billboard-points):2
  FID (Integer64) = 2
  POINT Z (1000 1000 200)
```
